pkg/lib/csv/csv_test.go

```diff
@@ -0,0 +1,44 @@
+package csv
+
+import (
+	"testing"
+)
+
+// TestCSVToStringArrayMap checks if trailing spaces are preserved in type
+// `string,array`.
+func TestCSVToStringArrayMap(t *testing.T) {
+	csv := `id,notes
+int,"string,array"
+0,"Trailing ,space"
+1," Leading,space"
+2," Surrounding ,space"`
+
+	have, err := CSVToMap([]byte(csv), ',')
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var want []map[string]interface{}
+	want = append(want,
+		map[string]interface{}{
+			"id":      0,
+			"notes": []string{"Trailing ", "space"},
+		},
+		map[string]interface{}{
+			"id":      1,
+			"notes": []string{" Leading", "space"},
+		},
+		map[string]interface{}{
+			"id":      2,
+			"notes": []string{" Surrounding ", "space"},
+		},
+	)
+
+	for i := range have {
+		for j := range have[i]["notes"].([]string) {
```

Comment:
whats that? use a normal for loop

```
for _, row := range have {
...
}
```

Reply:
Test required `row` of both `have` and `want` so went with `index`